src/diffusers/models/attention_dispatch.py

```diff
@@ -207,6 +210,22 @@ def list_backends(cls):
         return list(cls._backends.keys())
 
 
+@lru_cache(maxsize=None)
+def _load_fa3_hub():
+    from ..utils.kernels_utils import _get_fa3_from_hub
+
+    fa3_hub = _get_fa3_from_hub()  # won't re-download if already present
+    if fa3_hub is None:
+        raise RuntimeError(
+            "Failed to load FlashAttention-3 kernels from the Hub. Please ensure the wheel is available for your platform."
+        )
+    return fa3_hub
+
+
+def flash_attn_3_hub_func(*args, **kwargs):
+    return _load_fa3_hub().flash_attn_func(*args, **kwargs)
```

Comment:
Rather than loading the kernel every time we invoke flash attention, it would be better to import the function at the top of the file, similar to the other FA backends. 

https://github.com/huggingface/diffusers/blob/e58711e73cba70c1c02aaa67f80945a1458901b8/src/diffusers/models/attention_dispatch.py#L63-L68

Reply:
Sure but see
https://github.com/huggingface/diffusers/pull/12236#discussion_r2300940273

We shouldn’t make remote calls to the Hub unless requested.

Reply:
God that simplifies a lot of stuff. Thanks, Dhruv!